src/modm/ui/color/rgb565.hpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2009-2013, Fabian Greif
+ * Copyright (c) 2021, Thomas Sommer
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_COLOR_RGB565_HPP
+#define MODM_COLOR_RGB565_HPP
+
+#include <modm/platform.hpp>
+
+namespace modm
+{
+namespace color
+{
+
+template<typename T>
+class RgbT;
+
+template<typename T>
+class HsvT;
+
+// Rgb565 (565) Format
+/// @ingroup modm_ui_color
+class Rgb565
+{
+public:
+	uint16_t color{0x0000};
+
+	constexpr Rgb565() = default;
+
+	constexpr Rgb565(uint16_t color) : color(color) {}
+
+	constexpr Rgb565(uint8_t red, uint8_t green, uint8_t blue)
+		: color(uint16_t(red & 0xF8) << 8 | uint16_t(green & 0xFC) << 3 | uint16_t(blue >> 3))
+	{}
+
+	template<typename _T>
+	constexpr Rgb565(const RgbT<_T> &rgb) : Rgb565(rgb.red, rgb.green, rgb.blue)
+	{}
+
+	template<typename _T>
+	constexpr Rgb565(const HsvT<_T> &hsv) : Rgb565(RgbT<uint8_t>(hsv))
+	{}
+
+	inline bool
+	operator==(const Rgb565 &other) const
+	{
+		return (color == other.color);
+	}
```

Comment:
`constexpr` missing and the compiler can generate that one for you since C++20:
```suggestion
	constexpr bool
  	operator==(Rgb565 other) const = default;
```

Reply:
Sweeet